Feat/4 회원 조회(admin)

shushuaka opened the pull request:
### 구현 내용
**1. 회원 조회 기능 추가**
- GET /api/admin/users 엔드포인트 구현
- 관리자가 RoleType이 USER인 회원 목록을 페이지 단위로 조회할 수 있도록 서비스와 컨트롤러 개발
- UserAdminService를 통해 비즈니스 로직 분리

**2. 페이징 처리**
- PageRequest를 활용하여 회원 목록을 페이지 단위로 반환
- 응답 데이터에 현재 페이지, 전체 페이지 수, 페이지 당 회원 수 포함

**3. DTO 설계**
- UserAdminResponseDto: HTTP 상태 코드, 메시지, 회원 데이터 리스트, 페이지 정보를 포함한 응답 DTO 생성
- UserResponseDto: 회원 개별 정보(id, email, name, nickname, address, imagePath, role)를 포함한 DTO 추가

**4. 유효성 검증**
- page와 size 파라미터에 대한 기본값 및 1 이상 입력 여부 검증

**5. 예외 처리**
- 유효하지 않은 페이지 번호나 페이지 크기 입력 시 적절한 예외 메시지 반환

**Postman 테스트 완료**
- 로컬 환경에서 Postman을 사용하여 GET /api/admin/users 요청 테스트 완료
- 페이지 번호 및 크기에 따른 결과 확인
- 데이터가 존재하는 경우: 회원 목록 반환 확인
- 데이터가 없는 페이지 요청 시 빈 목록 반환 확인

@gaeul79 @Dohyeon-Parrk @gyeonwoo-jerry @dameun0527

shushuaka committed:
feat: User 엔티티 수정 및 imagePath 필드 추가 #4

- User 엔티티에 imagePath 필드 추가
- email 필드에 unique 제약 조건 추가
- orphanRemoval=true 설정 추가로 고아 객체 자동 삭제
- updateProfile 메서드 추가로 사용자 정보 업데이트 로직 구현

shushuaka committed:
feat: UserAdminController 생성 및 회원 조회 API 구현 #4

- UserAdminController 클래스 추가
- 회원 조회(ADMIN) API 엔드포인트 `/api/admin/users` 구현
- UserAdminService와 연동하여 관리자용 회원 조회 기능 제공

shushuaka committed:
feat: UserAdminResponseDto 생성 #4

- 관리자 회원 조회 API 응답 DTO 생성
- 회원 정보 리스트, HTTP 상태 코드, 응답 메시지, 페이징 정보를 포함

shushuaka committed:
feat: UserAdminService 구현 #4

- 관리자 회원 조회 기능 서비스 계층 구현
- RoleType.USER로 필터링된 회원 목록 조회 및 페이징 처리
- UserResponseDto로 사용자 정보 변환 후 UserAdminResponseDto 반환
- 입력값 유효성 검증 추가 (페이지 번호와 크기 1 이상 조건)

shushuaka committed:
refactor: UserRepository에 페이징 처리 메서드 추가 및 리팩터링 #4

- findAllByRole 메서드에 Pageable 지원 추가로 페이징 처리 구현
- 코드 가독성을 위해 주석 추가 및 불필요한 import 제거

shushuaka committed:
feat: UserResponseDto 추가 및 User 엔티티 변환 메서드 구현 #4

- User 엔티티 데이터를 UserResponseDto로 변환하는 fromEntity 메서드 추가
- 회원 정보를 클라이언트에 전달하기 위한 DTO 클래스 생성

shushuaka committed:
feat: UserLoginRequestDto 및 UserLoginResponseDto 수정 #4

- UserLoginRequestDto 구현을 위한 기본 구조 추가
- UserLoginResponseDto에 필요한 필드(token, email, name, role) 및 Builder 패턴 추가
- 회원 로그인 요청 및 응답 처리를 위한 DTO 클래스 완성

shushuaka committed:
feat: UserSignupRequestDto 및 UserSignupResponseDto 구현 #4

- UserSignupRequestDto에 입력값 유효성 검증을 위한 어노테이션 추가
- UserSignupResponseDto에 회원 가입 성공 시 반환할 정보(id, email, name, role) 추가

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into feat/4-회원-조회(ADMIN)

shushuaka committed:
refactor: GlobalExceptionHandler 리팩토링 및 코드 간소화 #4

- Validation 실패 예외 처리 메서드 개선: 에러 메시지 추출 로직 간소화 및 Helper Method 도입
- 비즈니스 로직 예외 처리 메서드 개선: 코드 재사용성 강화
- 기타 예외 처리 메서드 개선: 상세 로그 추가 및 Helper Method로 분리
- Helper Method 추가: buildErrorResponse 및 extractErrorMessage
- 중복 코드 제거 및 가독성 향상

dameun0527 commented on src/main/java/com/sparta/kidscafe/domain/user/controller/UserAdminController.java:
```diff
@@ -0,0 +1,27 @@
+
+    @GetMapping("/api/admin/users")
+    public ResponseEntity<UserAdminResponseDto> getAdminUsers(
+            @RequestHeader("Authorization") String authorization,
```
@Auth AuthUser authUser 매개변수 수정

Dohyeon-Parrk commented on src/main/java/com/sparta/kidscafe/domain/user/controller/UserAdminController.java:
AuthUser 사용하시는거 지수님께 설명 잘 들으셨죠??

shushuaka committed:
refactor: PasswordEncoder를 CustomPasswordEncoder로 이름 변경 및 관련 설정 수정

- Spring Security의 PasswordEncoder 인터페이스와 충돌을 방지하기 위해 PasswordEncoder 클래스 이름을 CustomPasswordEncoder로 변경
- PasswordConfig에서 CustomPasswordEncoder를 비밀번호 인코딩 빈으로 설정
- 프로젝트 전반에서 CustomPasswordEncoder를 일관되게 사용하도록 수정

shushuaka committed:
feat: User 엔티티에 비밀번호 변경 메서드 추가 및 유효성 검사 개선 #4

- User 엔티티에 비밀번호 변경을 위한 changePassword 메서드 추가
- updateProfile 메서드에서 빈 값이 업데이트되지 않도록 유효성 검사 강화
- 최소 생성자를 추가하여 User 생성 시 필요한 필드만 포함하도록 수정

shushuaka committed:
refactor: PasswordEncoder 리팩토링 및 CustomPasswordEncoder로 이름 변경

shushuaka committed:
refactor: UserAdminController와 UserAdminService 응답 DTO를 common 패키지로 통일 #4

- UserAdminController에서 UserAdminService 반환 타입을 ListResponseDto<UserResponseDto>로 변경
- UserAdminService가 UserAdminResponseDto 대신 ListResponseDto를 반환하도록 수정
- 공통 DTO인 ListResponseDto를 사용하여 프로젝트 응답 구조를 통일
- 컨트롤러와 서비스 간 타입 불일치 문제 해결

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into feat/4-회원-조회(ADMIN)

shushuaka committed:
Merge remote-tracking branch 'origin/dev' into feat/4-회원-조회(ADMIN)

shushuaka committed:
feat: UserAdminController에 @Auth 추가 및 관리자 권한 검증 로직 구현 #4

- @Auth 어노테이션을 사용하여 AuthUser 객체를 컨트롤러에 주입

gaeul79 reviewed (approved):
고생하셧습니다

shushuaka committed:
feat: UserAdminController 관리자 권한 예외 처리 로직 변경 #4

- IllegalArgumentException 대신 BusinessException 사용으로 예외 처리 로직 변경
- ErrorCode.FORBIDDEN을 활용하여 권한 부족 시 일관된 글로벌 예외 처리 적용
